### ISSUE TITLE
nvidia-container-toolkit version mismatch with nvidia driver 

### ISSUE THREAD

**Issue by antpanx** [void-linux/void-packages#51518]
### Is this a new report?

Yes

### System Info

Void 6.6.43_1 x86_64 AuthenticAMD uptodate rrmFFFFFFFFFFFF

### Package(s) Affected

libnvidia-container-1.13.5_1 , nvidia-container-toolkit-1.13.5_1, nvidia-550.100_1 , nvidia-dkms-550.100_1 , docker-27.1.1_1  

### Does a report exist for this bug with the project's home (upstream) and/or another distro?

Currently this seems to be a pure void-linux issue, as I suspect a versioning mismatch.

### Expected behaviour

sudo docker run --rm --runtime=nvidia --gpus all ubuntu nvidia-smi

It should return the same output as nvidia-smi on the host machine.

 *  nvidia-smi works as expected on the host machine.
 *  I ran sudo nvidia-ctk runtime configure --runtime=docker to make sure the packages are correctly configured. Although this shouldn't be necessary and already being ran by the package manager.

### Actual behaviour

Instead the following error is being returned:
  

> docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy'
>   nvidia-container-cli: detection error: open failed: /usr/lib/libnvidia-nvvm.so.550.78: no such file or directory: unknown.

It is clear this is caused by a version mismatch as "ls /usr/lib/libnvidia*" returns the following:

> [...]
> /usr/lib/libnvidia-nvvm.so.550.100
> [...]

(Output truncated, but is clear that everything has minor-version 100 instead of 78, which is inline with the current driver package)

I have an nvidia geforce rtx 4070 running on my system.

If someone has an nvidia-system as well, would be great to see if others have the same issue, or if I messed up my system somehow.

I have void-packages installed, feel free to guide me in the correct direction, if it is indeed a version mismatch within the packages, and I might be able to update the version-string to the correct version myself.

### Steps to reproduce

1. Use a system with nvidia-gpu (yeah sorry I need CUDA ;-) )
2. Install proprietary nvidia-driver + docker + nvidia-container-runtime. Most people just install most of the "nvidia-*"-packages. (No rootless docker to make our lives easier)
3. Try to run a basic workload ( c.f. [nvidia-docs](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/sample-workload.html)), such as the following
  > sudo docker run --rm --runtime=nvidia --gpus all ubuntu nvidia-smi

**Comment by antpanx** [comment]
Trying to update both libnvidia-container and nvidia-container-toolkit proved to be more complex than expected.

Obviously the rather "hacky" workaround
> sudo ln -s /usr/lib/libnvidia-nvvm.so.550.100 /usr/lib/libnvidia-nvvm.so.550.78

temporarily solves the problem for me right now. 

Perhaps some more experienced void-maintainer can guide me to a long-term solution or confirm if it is an issue specific to my installation.

**Comment by ahesford** [comment]
You probably need to reboot your system to load whatever version of the NVIDIA kmod matches your installed userspace tools.

**Comment by antpanx** [comment]
@ahesford Thanks for the help. I now restarted my workstation again and the error reappeared but now with a different version, apparently due to another update in the mean-time ;-).

![Screenshot_void_nvidia_container_cli_versions](https://github.com/user-attachments/assets/81e18512-762d-4eec-bbe8-635c2c662696)

I restarted my pc multiple times and also ran "sudo nvidia-ctk runtime configure --runtime=docker" again (with another restart).

Right now I'll assume it only affects my computer due to some strange configuration issue. For now I will keep symlinking as a temporary workaround until I have the time to do a separate reinstallation and see it it reappears.

Therefore **closing** this issue to save maintainer resources.  Thanks for the suggestions so far!
